src/notifications/NotificationsManager.ts

```diff
@@ -161,26 +353,210 @@ class NotificationsManager {
    */
   @ready(new notificationsErrors.ErrorNotificationsNotRunning())
   public async sendNotification(
-    nodeId: NodeId,
-    data: NotificationData,
-  ): Promise<void> {
-    const notification: Notification = {
+    {
+      nodeId,
+      data,
+      retries = 64,
+    }: {
+      nodeId: NodeId;
+      data: NotificationData;
+      retries?: number;
+    },
+    tran?: DBTransaction,
+  ): Promise<{
+    notificationId: NotificationId;
+    sendProm: Promise<void>;
+  }> {
+    const nodeIdEncoded = nodesUtils.encodeNodeId(nodeId);
+    const notificationId = this.outboxNotificationIdGenerator();
+    const notificationIdEncoded =
+      notificationsUtils.encodeNotificationId(notificationId);
+    const notificationDb: NotificationDB = {
       typ: 'notification',
       data: data,
       isRead: false,
       iss: nodesUtils.encodeNodeId(this.keyRing.getNodeId()),
-      sub: nodesUtils.encodeNodeId(nodeId),
+      sub: nodeIdEncoded,
     };
-    const signedNotification = await notificationsUtils.generateNotification(
-      notification,
-      this.keyRing.keyPair,
+    let pendingTask: Task | undefined;
+    const transactionCallback = async (tran: DBTransaction) => {
+      await tran.put(
+        [
+          ...this.notificationsManagerOutboxNotificationsDbPath,
+          idUtils.toBuffer(notificationId),
+        ],
+        notificationDb,
+      );
+      const counter = await tran.get<number>(
+        this.notificationsManagerOutboxNotificationsCounterDbPath,
+      );
+      await tran.put(
+        this.notificationsManagerOutboxNotificationsCounterDbPath,
+        counter != null ? counter + 1 : 1,
+      );
+      pendingTask = await this.taskManager.scheduleTask(
+        {
+          handlerId: this.sendNotificationHandlerId,
+          parameters: [
+            {
+              nodeIdEncoded,
+              notificationIdEncoded,
+              retries,
+              attempt: 0,
+            },
+          ],
+          path: [this.sendNotificationHandlerId],
+          lazy: false,
+        },
+        tran,
+      );
+      this.outboxNotificationIdEncodedToTaskIdMap.set(
+        notificationIdEncoded,
+        pendingTask.id,
+      );
+    };
+    if (tran == null) {
+      await this.db.withTransactionF(transactionCallback);
+    } else {
+      await transactionCallback(tran);
+    }
```

Comment:
We use recursion to ensure a transaction is created when it's not provided. Do something like this to the top of a method.

```ts
  public async setNode(
    nodeInfo: GestaltNodeInfo,
    tran?: DBTransaction,
  ): Promise<['node', NodeId]> {
    if (tran == null) {
      return this.db.withTransactionF((tran) => this.setNode(nodeInfo, tran));
    }
    //...
  }
```

Reply:
i've updated this to follow the suggestion